dvc/utils/fs.py

```diff
@@ -61,14 +62,14 @@ def __init__(self, path, base_path):
 
 
 def contains_symlink_up_to(path, base_path):
-    if base_path not in path:
+    if fspath(base_path) not in fspath(path):
```

Comment:
This is the case where you could simply:
```suggestion
    base_path = fspath(base_path)
    path = fspath(path)
    
    if fspath(base_path) not in fspath(path):
```

Reply:
@efiop Yes! I realized it was failing some possibilities too. :)